File: Pet_Shop/wwwroot/js/models/Produto.js

export default class Produto {

    constructor(produto) {
        this.cod = produto.cod || null;//garanti que tds propriedades tenha um valor msm que null
        this.nome = produto.nome || null;
        this.valor = produto.valor || null;
        this.quantidade = produto.quantidade || null;
        this.estoque_minimo = produto.estoque_minimo || null;
        this.categoria = produto.categoria || null;
        this.descricao = produto.descricao || null;
        this.imagem = produto.imagem || null;
    }

    get cod() {
        return this._cod;
    }
    set cod(novoCod) {
        this._cod = novoCod;
    }

    get nome() {
        return this._nome;
    }
    set nome(novoNome) {
        this._nome = novoNome;
    }

    get valor() {
        return this._valor;
    }
    set valor(novoValor) {
        this._valor = novoValor;
    }

    get quantidade() {
        return this._quantidade;
    }
    set quantidade(novoQuantidade) {
        this._quantidade = novoQuantidade;
    }

    get estoque_minimo() {
        return this._estoque_minimo;
    }
    set estoque_minimo(novoEstoque_minimo) {
        this._estoque_minimo = novoEstoque_minimo;
    }

    get categoria() {
        return this._categoria;
    }
    set categoria(novoCategoria) {
        this._categoria = novoCategoria;
    }

    get descricao() {
        return this._descricao;
    }
    set descricao(novoDescricao) {
        this._descricao = novoDescricao;
    }

    get imagem() {
        return this._imagem;
    }
    set imagem(novoImagem) {
        this._imagem = novoImagem;
    }
}

File: Pet_Shop/Models/Produto.cs
using Microsoft.AspNetCore.Mvc.ModelBinding;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Pet_Shop.Models
{
    public class Produto
    {
        [Key]
        public int Cod { get; set; }

        [Required(ErrorMessage = "O Nome deve ser informado")]
        [StringLength(30, MinimumLength = 3, ErrorMessage = "O nome deve ter de 3 à 30 caracteres")]
        public string Nome { get; set; }

        [Required(ErrorMessage = "O Valor deve ser informado")]
        [Range(0, int.MaxValue, ErrorMessage = "O valor deve ser igual ou maior que 0")]
        public double Valor { get; set; }


        [Required(ErrorMessage = "A Quantidade deve ser informado")]
        [Range(0, int.MaxValue, ErrorMessage = "A Quantidade deve ser igual ou maior que 0")]
        public int Quantidade { get; set; }

        [Required(ErrorMessage = "O Estoque Minimo deve ser informado")]
        [Range(0, int.MaxValue, ErrorMessage = "O Estoque Minimo deve ser igual ou maior que 0")]
        public int Estoque_Minimo { get; set; }

        [Required(ErrorMessage = "A Categoria deve ser informado")]
        public string Categoria { get; set; }

        [Required(ErrorMessage = "A Descrição deve ser informada")]
        [StringLength(50, MinimumLength = 5, ErrorMessage = "O Estoque Minimo deve ter de 5 à 50 caracteres")]
        public string Descricao { get; set; }

        [BindNever]
        public byte[] Imagem{ get; set; }
    }
}
